### ISSUE TITLE
Countdown timer incorrectly shows extra time remaining after computer sleep

### ISSUE THREAD

**Issue by daprice** [ludumdare/ludumdare#1431]
The countdown timer displays the wrong amount of time remaining if the page is left open and the execution is paused for whatever reason (in my case, I think it’s because my computer went to sleep, but some browsers may pause JS execution in background tabs so that could also potentially trigger this).

Steps to reproduce (Safari 11.0.1 on MacOS):
1. Open Ludum Dare website during an event with a countdown timer
2. Put computer in sleep/hibernate/standby
3. Wake computer up after some time has passed

Expected result: timer shows the correct amount of time remaining until the end of the event
Actual result: timer keeps counting down but is offset by the length of time the computer was asleep, so if the computer was asleep for an hour, it would show an extra hour

**Comment by povrazor** [comment]
Might be a mac/safari-only issue. I sleep my laptop all the time, and haven't seen this.

**Comment by zwrawr** [comment]
I'm guessing the solution is to not assume each update is one second here https://github.com/ludumdare/ludumdare/blob/master/src/com/clock-flip/flip.js#L42 .

Or ask the server for the correct time every once in a while.

**Comment by mikekasprzak** [comment]
Good catch. I assumed it was a subtraction going on there, but it looks
like it's actually a decrement.

Mike Kasprzak

**Comment by local-minimum** [comment]
After checking through the code I'd say the only reasonable thing is to rewrite most of it from scratch, except perhaps for the animation details.

**Comment by StormBurpee** [comment]
Or alternatively, we could store a variable for the last system time the clock updated (maybe do this every 10 iterations?) Then if there's a large difference between the last update time and the new update time, force a resync? 
That's a much quicker fix than rewriting the entire class.
